### PR TITLE
Add blog post for deprecation of the dependency api

### PR DESCRIPTION
**Pull request by segiddins**

### CODE REVIEW

**Comment by simi** [comment]
:thinking: What's the proper name to refer to https://rubygems.org as a service? It seems both `rubygems.org` and `RubyGems.org` are used in the text.

File: _posts/2023-02-22-dependency-api-deprecation.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+The dependency API makes up a huge percentage of the traffic served by by the rubygems.org rails app, and is one of the few high-traffic endpoints that is not cachable by fastly.
+
+The Compact Index was designed to be completely cacheable and served at the edge by Fastly. In contrast, the dependency API is completely _uncacheable_, and requires RubyGems.org to run a complex database query for each request, accounting for almost 50% of the total uncached request volume to RubyGems.org with nearly 3000qps.
+
+Because the Compact Index has been out for over half a decade and Bundler & RubyGems both support fallback to the "full" index (`specs.4.8.gz` files) when the Dependency API is unavailable, we plan to deprecate and delete the dependency API from rubygems.org over the coming months.
+
+### What impact will this deprecation have?
+
+If you're running a version of bundler >= 1.12, there should be no impact. Bundler sequently checks the compact index API, the dependency API, and finally the full API. Versions of bundler that support the compact index API don't use the dependency API at all for gem sources
```
Comment (simi) [inline _posts/2023-02-22-dependency-api-deprecation.md:28]: What about to include also the related Ruby/release date of bundler 1.12 (seems like Ruby 1.8 was related,https://rubygems.org/gems/bundler/versions/1.12.0)? That would make it clear it is ~7 years old release if I remember well.

File: _posts/2023-02-22-dependency-api-deprecation.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+that serve the compact index, including RubyGems.org.
```
Comment (simi) [inline _posts/2023-02-22-dependency-api-deprecation.md:29]: Would it be possible to explicitly mention non-rubygems.org hosts (usually self-hosted private gems source like gemstash, geminabox, nexus and commercial private hosts github packages, gemfury...) are not affected?

File: _posts/2023-02-22-dependency-api-deprecation.md (hunk 1 above)
Comment (indirect) [reply _posts/2023-02-22-dependency-api-deprecation.md:28]: A good idea :) I will add the date 👍🏻

**Comment by indirect** [comment]
Updated 👍🏻

**Review by indirect — APPROVED** [review]

**Review by simi — APPROVED** [review]

**Comment by deivid-rodriguez** [comment]
This is so awesome!! 🎉

File: _posts/2023-02-22-dependency-api-deprecation.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+In contrast to the Dependency API, the Compact Index API is designed to be cachable. Since the  Compact Index has been out for almost seven years, and since Bundler & RubyGems both support fallback to the slower "full" index when the Dependency API is unavailable, we plan to deprecate and delete the dependency API from RubyGems.org over the coming months.
+
+### What impact will this deprecation have?
+
+We expect this deprecation to have no user-facing impact.
+
+If you are installing gems from RubyGems.org using versions of Bundler more than 7 years old (1.11 or older), your installs will get slower. Hopefully, everyone has upgraded their copy of Bundler at least once in the last 7 years, and their installs can continue at full speed.
+
+If you are installing gems from other gem sources, like [Gemstash][1], [Geminabox][2], [Nexus][3], [Artifactory][4], or [GitHub Packages][5], nothing will change. Bundler will continue to try the Compact Index API first, then fall back to the Dependency API, then fall back to the full index.
```
Comment (kzaitsev) [inline _posts/2023-02-22-dependency-api-deprecation.md:32]: unfortunately, change https://issues.sonatype.org/browse/NEXUS-38452?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedId=1257013#comment-1257013

Nexus and artifactory didn't implement compact API, with some tweaks they will be worked in full index mode.

**Pull request merged**